operator/pkg/values/translation/translation.go

```diff
@@ -161,17 +167,38 @@ func SetScannerDBValues(sv *ValuesBuilder, db *platform.DeploymentSpec) {
 }
 
 // SetScannerV4DBValues sets values in "sv" based on "db"
-func SetScannerV4DBValues(sv *ValuesBuilder, db *platform.ScannerV4DB) {
+// Unlike central-db's PVC we don't use the extension.ReconcilePVCExtension.
+// The operator creates this PVC through the helm chart. This means it is managed
+// by the default helm lifecycle, instead of the operator extension. The difference is
+// that the extension prevents central DB's PVC deletion on deletion of the CR.
+// Since scanner V4's DB contains data which recovers by itself it is safe to remove the PVC
+// through the helm uninstall if a CR is deleted.
+func SetScannerV4DBValues(ctx context.Context, sv *ValuesBuilder, db *platform.ScannerV4DB, objKind string, client ctrlClient.Client) {
+	dbVB := NewValuesBuilder()
+	persistenceVB := NewValuesBuilder()
+
+	useEmptyDir, err := shouldUseEmptyDir(ctx, objKind, db, client)
+	if err != nil {
+		sv.SetError(fmt.Errorf("error : %w", err))
+		return
+	}
+
+	if useEmptyDir {
+		persistenceVB.SetBoolValue("none", true)
+		dbVB.AddChild("persistence", &persistenceVB)
+		sv.AddChild("db", &dbVB)
+		return
+	}
+
 	if db == nil {
 		return
 	}
 
-	dbVB := NewValuesBuilder()
 	dbVB.SetStringMap("nodeSelector", db.NodeSelector)
 	dbVB.AddChild(ResourcesKey, GetResources(db.Resources))
 	dbVB.AddAllFrom(GetTolerations(TolerationsKey, db.Tolerations))
 	setScannerV4DBPersistence(&dbVB, db.Persistence)
```

Comment:
Please first read my longer comment to operator/pkg/values/translation/translation_test.go.

--

I guess here would be the place to add set something like `createClaim = true`.

Reply:
See above.